docs/usage/shoot_high_availability_best_practices.md

```diff
@@ -0,0 +1,431 @@
+
+- If the `kubelet` restarts, it will report all pods as `NotReady` on startup until it reruns its probes ([#100277](https://github.com/kubernetes/kubernetes/issues/100277)), which leads to temporary endpoint and load balancer target removal ([#102367](https://github.com/kubernetes/kubernetes/issues/102367)). Pull requests to improve from our colleagues were rejected. [Tim Hockin](https://www.linkedin.com/in/tim-hockin-6501072) joined the discussion as well, criticizing the current behavior, but so far to no avail. Gardener uses rolling updates and a jitter to spread necessary `kubelet` restarts as good as possible, but this `kubelet` logic is suboptimal and unpredictable.
+- If a `kube-proxy` pod on a node turns `NotReady`, all load balancer traffic to all pods (on this node) under services with `externalTrafficPolicy` `local` will cease as the load balancer will then take this node out of serving. This makes no sense and was discussed by us in [Slack Kubernetes #sig-network](https://kubernetes.slack.com/archives/C09QYUH5W/p1642764716200400), but the maintainers remain reluctant (like above) to change the current behavior (even though nobody could remember why this was ever done or deliver a reason why it may be a good thing to keep). So, please remember that `externalTrafficPolicy` `local` not only has the disadvantage of imbalanced traffic spreading, but also a dependency to the kube-proxy pod that may and will be unavailable during updates. Gardener uses rolling updates to spread necessary `kube-proxy` updates as good as possible, but this `kube-proxy` logic is suboptimal and unpredictable.
+
+These things of note are just a start and by no means complete. They may or may not affect you, but other intricacies may. It's a reminder to be watchful as Kubernetes may have one or two relevant quirks that you need to consider (and will probably only find out over time and with extensive testing).
+
+## Meaningful Availability
+
+Finally, let's go back to where we started. We recommended to measure the [*meaningful availability*](https://research.google/pubs/pub50828) (as opposed to the *meaningless availability*, a term coined by a great colleague of mine who introduced me to the subject matter). In Gardener, we cannot do that to the extent that we'd like to as the means of interaction with Gardener and its clusters is `kubectl` that we do not control. Plus, it remains difficult to tell apart "user-local phenomena" from the service as such. Still, we do not trust only internal signals, but track also whether Gardener or the control planes that it manages are externally available through the external DNS records and load balancers, SNI-routing Istio gateways, etc. (the same path all users must take). It's a huge difference whether the API server's internal readiness probe passes or the user can actually reach the API server and it does what it's supposed to do. Most likely, you will be in a similar spot and can do the same.
```

Comment:
>  In Gardener, we cannot do that to the extent that we'd like to as the means of interaction with Gardener and its clusters is `kubectl` that we do not control.

Should we mention this here at all? To me it's very unspecific and I'm not aware of the importance to the reader.

Reply:
Thank you very much, agreed, I cut that paragraph down significantly and kept only enough to mention DWD (not so much that end users leverage it, but as an example that internal probes are not always sufficient).